pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -1574,6 +1574,81 @@ func parseU32ConfigItem(newGlobalConfigPtr *types.GlobalConfig, key string,
 
 var itemsPrevConfigHash []byte
 
+type ConfigItemType uint8
+
+const (
+	CONFIGITEM_TYPE_INT ConfigItemType = iota
+	CONFIGITEM_TYPE_BOOL
+	CONFIGITEM_TYPE_STRING
+	CONFIGITEM_TYPE_TRISTATE
+)
+
+type ConfigItemSpec struct {
+	Key      string
+	ItemType ConfigItemType
+
+	intMin     int32
```

Comment:
Given that golang doesn't have a discriminated union I think we should at least add comments to indicate that these int* fields are only used for the int ItemType, and the xxxDefault are only used if the xxx ItemType